src/main/java/org/zalando/riptide/RestBase.java

```diff
@@ -77,4 +88,43 @@ public D execute(final HttpMethod method, final URI url, final HttpHeaders heade
 
     protected abstract <T> D execute(final HttpMethod method, final URI url, final HttpEntity<T> entity);
 
+    static <T> void writeRequestEntity(final HttpEntity<T> entity, final ClientHttpRequest request, List<HttpMessageConverter<?>> converters) throws IOException {
+        final HttpHeaders headers = entity.getHeaders();
+        request.getHeaders().putAll(headers);
+
+        @Nullable final T body = entity.getBody();
+
+        if (body != null) {
+
+            final Class<?> type = body.getClass();
+            @Nullable final MediaType contentType = headers.getContentType();
+
+            final Optional<HttpMessageConverter<T>> match = converters.stream()
```

Comment:
> Should we sort them by specificity first?

Is there such a thing? I thought the order in this list should be the order they are tried.